File: src/EnglishGame/wwwroot/app/game/uduel.ts
import { URound } from './uround';
import { UUser } from './uuser';

export class UDuel {
    Id: number;
    USubjectId: number;
    PrimaryPlayerId: string;
    PrimaryPlayer: UUser;
    SecondaryPlayerId: string;
    SecondaryPlayer: UUser;
    Summary: string;
    URounds: URound[];
}

File: src/EnglishGame/Models/UDuel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EnglishGame.Models
{
    public class UDuel
    {
        public int Id { get; set; }

        public int USubjectId { get; set; }
        public USubject USubject { get; set; }

        public string PrimaryPlayerId { get; set; }
        public UUser PrimaryPlayer { get; set; }

        public string SecondaryPlayerId { get; set; }
        public UUser SecondaryPlayer { get; set; }

        public string Summary { get; set; }

        public List<URound> URounds { get; set; }
    }
}
